File: 812-rotate-string/rotate-string.js
var rotateString = function (s, goal) {
    const m = s.length
    const n = goal.length
    if (m !== n) return false;
    let arr = s.split('')
    for (let rotationCount = 1; rotationCount <= m; rotationCount++) {
        const firstChar = arr.shift()
        arr.push(firstChar)
        if (arr.join('') === goal) return true
    }
    return false
};

File: 812-rotate-string/rotate-string.cpp
class Solution {
public:
    bool rotateString(string s, string goal) {
        int m = s.length();
        int n = goal.length();
        if (m != n)
            return false;
        for (int rotationCount = 1; rotationCount <= m; rotationCount++) {
            // rotate(start of range, which position will become start position of range, end of range)
            rotate(s.begin(), s.begin() + 1, s.end());
            if (s == goal)
                return true;
        }
        return false;
    }
};